src/Archive/Adapter/Gzip.php

```diff
@@ -84,7 +84,7 @@ public function compress($files, string $destination = '', bool $overwrite = fal
         }
 
         $filename = $destination;
-        if ($outfile = gzopen($filename, $mode)) {
+        if ($outfile = gzopen($filename, 'wb'.$mode)) {
```

Comment:
I think  it should verify the compression mode before calling the `gzopen` function.

According to the [gzcompress](https://www.php.net/manual/en/function.gzcompress.php) reference, the compression level range should be from `0` to `9`.

Reply:
> I think it should verify the compression mode before calling the `gzopen` function.

agreed :+1: